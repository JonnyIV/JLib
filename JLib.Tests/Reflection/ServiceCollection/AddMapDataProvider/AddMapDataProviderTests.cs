

















// disabled since the mapping is being reworked

















//using Microsoft.Extensions.DependencyInjection;
//using System;
//using System.Collections.Generic;
//using System.Linq;
//using System.Text;
//using System.Threading.Tasks;
//using AutoMapper;
//using JLib.Data;
//using JLib.FactoryAttributes;
//using JLib.Helper;
//using JLib.HotChocolate;
//using JLib.Tests.Reflection.ServiceCollection.AddDataProvider;
//using Moq;
//using Xunit.Abstractions;

//namespace JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider;

//public class AddMapDataProviderTests : ReflectionTestBase
//{
//    public class TestArguments : ReflectionTestArguments
//    {
//        protected override string Filter { get; } = "";

//        protected override IEnumerable<ReflectionTestOptions> Options { get; } = new ReflectionTestOptions[]
//        {
//            new(
//                "srv_all_ce",
//                new []
//                {
//                    "input: xrm, ce",
//                    "overload: filter by tvt",
//                    "repos: none",
//                    "filter: MappedCe",
//                    "expected behavior: map from ce to xrm"
//                },
//                new []
//                {
//                    typeof(TestXrmEntity),
//                    typeof(TestCommandEntity),
//                },
//                (services, cache, exceptions)
//                    =>services.AddMapDataProvider<MappedCommandEntityType>(cache,_=>true,
//                        tvt=>tvt.SourceEntity,tvt=>!tvt.ReverseMap,exceptions)
//                ),
//            new(
//                "srv_all_gdo",
//                new []
//                {
//                    "input: xrm, gdo",
//                    "overload: filter by tvt",
//                    "repos: none",
//                    "filter: MappedGdo",
//                    "expected behavior: map from ce to xrm"
//                },
//                new []
//                {
//                    typeof(TestXrmEntity),
//                    typeof(TestCommandEntity),
//                    typeof(TestGraphQlDataObject),
//                },
//                (services, cache, exceptions)
//                    =>services.AddMapDataProvider<MappedGraphQlDataObjectType>(cache,_=>true,
//                        tvt=>tvt.SourceEntity,tvt=>!tvt.ReverseMap,exceptions)
//            ),
//            new(
//                "srv_all_allEntities_separateCall",
//                new []
//                {
//                    "input: all entities",
//                    "call: separate",
//                    "repos: none",
//                    "filter: MappedGdo",
//                    "expected behavior: map from ce to xrm"
//                },
//                new []
//                {
//                    typeof(TestXrmEntity),
//                    typeof(TestCommandEntity),
//                    typeof(TestGraphQlDataObject),
//                },
//                (services, cache, exceptions)
//                    =>services.AddMapDataProvider<MappedGraphQlDataObjectType>(cache,_=>true,
//                        tvt=>tvt.SourceEntity,tvt=>!tvt.ReverseMap,exceptions)
//                        .AddMapDataProvider<MappedCommandEntityType>(cache,_=>true,
//                            tvt=>tvt.SourceEntity,tvt=>!tvt.ReverseMap,exceptions)

//            ),
//            new(
//                "srv_all_allEntities_combinedCall",
//                new []
//                {
//                    "input: all entities",
//                    "repos: none",
//                    "filter: MappedGdo",
//                    "expected behavior: map from ce to xrm"
//                },
//                new []
//                {
//                    typeof(TestXrmEntity),
//                    typeof(TestCommandEntity),
//                    typeof(TestGraphQlDataObject),
//                },
//                (services, cache, exceptions)
//                    =>services.AddMapDataProvider(cache, exceptions)
//            ),
//            new(
//                "srv_all_allEntities_CombCall_allRep",
//                new []
//                {
//                    "input: all entities",
//                    "repos: none",
//                    "filter: MappedGdo",
//                    "Expected Implementations:",
//                    $"  {nameof(TestXrmEntity)}",
//                    $"    IDataProviderR       : {typeof(TestDataProvider<TestXrmEntity>).FullClassName()}",
//                    $"    IDataProviderRw      : {typeof(TestDataProvider<TestXrmEntity>).FullClassName()}",
//                    $"    ISourceDataProviderR : {typeof(TestDataProvider<TestXrmEntity>).FullClassName()}",
//                    $"    ISourceDataProviderRw: {typeof(TestDataProvider<TestXrmEntity>).FullClassName()}",
//                    $"  {nameof(TestCommandEntity)}",
//                    $"    IDataProviderR       : {nameof(TestCeRepository)}",
//                    $"    IDataProviderRw      : {nameof(TestCeRepository)}",
//                    $"    ISourceDataProviderR : {typeof(WritableMapDataProvider<TestXrmEntity,TestCommandEntity>).FullClassName()}",
//                    $"    ISourceDataProviderRw: {typeof(WritableMapDataProvider<TestXrmEntity,TestCommandEntity>).FullClassName()}",
//                    $"  {nameof(TestGraphQlDataObject)}",
//                    $"    IDataProviderR       : {nameof(TestGdoRepository)}",
//                    $"    IDataProviderRw      : -",
//                    $"    ISourceDataProviderR : {typeof(MapDataProvider<TestXrmEntity,TestGraphQlDataObject>).FullClassName()}",
//                    $"    ISourceDataProviderRw: -",
//                },
//                new []
//                {
//                    typeof(TestXrmEntity),
//                    typeof(TestCommandEntity),
//                    typeof(TestGraphQlDataObject),
//                    typeof(TestCeRepository),
//                    typeof(TestGdoRepository),
//                },
//                (services, cache, exceptions)
//                    =>services.AddMapDataProvider(cache, exceptions)
//                )
//        };
//    }
//    public AddMapDataProviderTests(ITestOutputHelper testOutput)
//        : base(testOutput)
//    {
//    }
//    [Theory, ClassData(typeof(TestArguments))]
//    public override void Test(ReflectionTestOptions options, bool skipTest)
//        => base.Test(options, skipTest);

//    public override void AddServices(IServiceCollection services, ITypeCache cache, IExceptionManager exceptions)
//        => services
//            .AddSingleton(new Mock<IMapper>().Object)
//            .AddSingleton(new Mock<IConfigurationProvider>().Object)
//            .AddRepositories(cache, exceptions)
//            .AddDataProvider<CommandEntityType, TestDataProvider<TestXrmEntity>, TestXrmEntity>(
//                cache, x => x.Value == typeof(TestXrmEntity), null, null, exceptions);

//    #region TestedClasses

//    public class TestDataProvider<T> : ISourceDataProviderRw<T>
//        where T : TestXrmEntity
//    {
//        public IQueryable<T> Get() => throw new NotImplementedException();

//        public void Add(T item) => throw new NotImplementedException();

//        public void Add(IEnumerable<T> items) => throw new NotImplementedException();

//        public void Remove(Guid itemId) => throw new NotImplementedException();

//        public void Remove(IEnumerable<Guid> itemIds) => throw new NotImplementedException();
//    }

//    public class TestXrmEntity : ICommandEntity
//    {
//        public Guid Id { get; }
//    }
//    public class TestCommandEntity : IMappedCommandEntity<TestXrmEntity>
//    {
//        public Guid Id { get; }
//    }
//    public class TestGraphQlDataObject : IMappedGraphQlDataObject<TestXrmEntity>
//    {
//        public Guid Id { get; }
//    }
//    public class TestCeRepository : IDataProviderRw<TestCommandEntity>
//    {
//        public IQueryable<TestCommandEntity> Get() => throw new NotImplementedException();

//        public void Add(TestCommandEntity item) => throw new NotImplementedException();

//        public void Add(IEnumerable<TestCommandEntity> items) => throw new NotImplementedException();

//        public void Remove(Guid itemId) => throw new NotImplementedException();

//        public void Remove(IEnumerable<Guid> itemIds) => throw new NotImplementedException();
//    }
//    public class TestGdoRepository : IDataProviderR<TestGraphQlDataObject>
//    {
//        public IQueryable<TestGraphQlDataObject> Get() => throw new NotImplementedException();
//    }
//    #endregion
//}
