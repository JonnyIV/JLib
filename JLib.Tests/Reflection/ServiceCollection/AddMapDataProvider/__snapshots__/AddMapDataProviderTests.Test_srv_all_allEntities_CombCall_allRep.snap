{
  "Parameters": {
    "TestName": "srv_all_allEntities_CombCall_allRep",
    "ExpectedBehavior": [
      "  input: all entities                                                                                                                     ",
      "  repos: none                                                                                                                             ",
      "  filter: MappedGdo                                                                                                                       ",
      "  Expected Implementations:                                                                                                               ",
      "    TestXrmEntity                                                                                                                         ",
      "      IDataProviderR       : AddMapDataProviderTests.TestDataProvider<AddMapDataProviderTests.TestXrmEntity>                              ",
      "      IDataProviderRw      : AddMapDataProviderTests.TestDataProvider<AddMapDataProviderTests.TestXrmEntity>                              ",
      "      ISourceDataProviderR : AddMapDataProviderTests.TestDataProvider<AddMapDataProviderTests.TestXrmEntity>                              ",
      "      ISourceDataProviderRw: AddMapDataProviderTests.TestDataProvider<AddMapDataProviderTests.TestXrmEntity>                              ",
      "    TestCommandEntity                                                                                                                     ",
      "      IDataProviderR       : TestCeRepository                                                                                             ",
      "      IDataProviderRw      : TestCeRepository                                                                                             ",
      "      ISourceDataProviderR : WritableMapDataProvider<AddMapDataProviderTests.TestXrmEntity, AddMapDataProviderTests.TestCommandEntity>    ",
      "      ISourceDataProviderRw: WritableMapDataProvider<AddMapDataProviderTests.TestXrmEntity, AddMapDataProviderTests.TestCommandEntity>    ",
      "    TestGraphQlDataObject                                                                                                                 ",
      "      IDataProviderR       : TestGdoRepository                                                                                            ",
      "      IDataProviderRw      : -                                                                                                            ",
      "      ISourceDataProviderR : MapDataProvider<AddMapDataProviderTests.TestXrmEntity, AddMapDataProviderTests.TestGraphQlDataObject>        ",
      "      ISourceDataProviderRw: -                                                                                                            "
    ],
    "includedTypes": [
      "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity",
      "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestCommandEntity",
      "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestGraphQlDataObject",
      "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestCeRepository",
      "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestGdoRepository"
    ]
  },
  "exception": null,
  "cache": {
    "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider": {
      "JLib.CommandEntityType": [
        "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity"
      ],
      "JLib.MappedCommandEntityType": [
        "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestCommandEntity"
      ],
      "JLib.MappedGraphQlDataObjectType": [
        "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestGraphQlDataObject"
      ],
      "JLib.RepositoryType": [
        "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestCeRepository",
        "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestGdoRepository"
      ]
    }
  },
  "services": {
    "non-generic": {
      "JLib.ITypeCache": [
        {
          "Lifetime": "Singleton",
          "ImplementationSource": "Instance",
          "ServiceType": "JLib.ITypeCache",
          "ServiceArguments": [],
          "ImplementationType": "JLib.TypeCache",
          "ImplementationTypeArguments": []
        }
      ],
      "AutoMapper.IMapper": [
        {
          "Lifetime": "Singleton",
          "ImplementationSource": "Instance",
          "ServiceType": "AutoMapper.IMapper",
          "ServiceArguments": [],
          "ImplementationType": "Castle.Proxies.IMapperProxy",
          "ImplementationTypeArguments": []
        }
      ],
      "AutoMapper.IConfigurationProvider": [
        {
          "Lifetime": "Singleton",
          "ImplementationSource": "Instance",
          "ServiceType": "AutoMapper.IConfigurationProvider",
          "ServiceArguments": [],
          "ImplementationType": "Castle.Proxies.IConfigurationProviderProxy",
          "ImplementationTypeArguments": []
        }
      ],
      "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestCeRepository": [
        {
          "Lifetime": "Scoped",
          "ImplementationSource": "Type",
          "ServiceType": "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestCeRepository",
          "ServiceArguments": [],
          "ImplementationType": "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestCeRepository",
          "ImplementationTypeArguments": []
        }
      ],
      "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestGdoRepository": [
        {
          "Lifetime": "Scoped",
          "ImplementationSource": "Type",
          "ServiceType": "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestGdoRepository",
          "ServiceArguments": [],
          "ImplementationType": "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestGdoRepository",
          "ImplementationTypeArguments": []
        }
      ]
    },
    "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestCommandEntity": {
      "JLib.Data.IDataProviderR<>": [
        {
          "Lifetime": "Scoped",
          "ImplementationSource": "Factory",
          "ServiceType": "JLib.Data.IDataProviderR<>",
          "ServiceArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestCommandEntity"
          ],
          "ImplementationType": "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestCeRepository",
          "ImplementationTypeArguments": []
        }
      ],
      "JLib.Data.IDataProviderRw<>": [
        {
          "Lifetime": "Scoped",
          "ImplementationSource": "Factory",
          "ServiceType": "JLib.Data.IDataProviderRw<>",
          "ServiceArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestCommandEntity"
          ],
          "ImplementationType": "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestCeRepository",
          "ImplementationTypeArguments": []
        }
      ],
      "JLib.Data.WritableMapDataProvider<>": [
        {
          "Lifetime": "Scoped",
          "ImplementationSource": "Type",
          "ServiceType": "JLib.Data.WritableMapDataProvider<>",
          "ServiceArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity",
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestCommandEntity"
          ],
          "ImplementationType": "JLib.Data.WritableMapDataProvider<>",
          "ImplementationTypeArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity",
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestCommandEntity"
          ]
        }
      ],
      "JLib.Data.ISourceDataProviderR<>": [
        {
          "Lifetime": "Scoped",
          "ImplementationSource": "Factory",
          "ServiceType": "JLib.Data.ISourceDataProviderR<>",
          "ServiceArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestCommandEntity"
          ],
          "ImplementationType": "JLib.Data.WritableMapDataProvider<>",
          "ImplementationTypeArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity",
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestCommandEntity"
          ]
        }
      ],
      "JLib.Data.ISourceDataProviderRw<>": [
        {
          "Lifetime": "Scoped",
          "ImplementationSource": "Factory",
          "ServiceType": "JLib.Data.ISourceDataProviderRw<>",
          "ServiceArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestCommandEntity"
          ],
          "ImplementationType": "JLib.Data.WritableMapDataProvider<>",
          "ImplementationTypeArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity",
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestCommandEntity"
          ]
        }
      ]
    },
    "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestGraphQlDataObject": {
      "JLib.Data.IDataProviderR<>": [
        {
          "Lifetime": "Scoped",
          "ImplementationSource": "Factory",
          "ServiceType": "JLib.Data.IDataProviderR<>",
          "ServiceArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestGraphQlDataObject"
          ],
          "ImplementationType": "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestGdoRepository",
          "ImplementationTypeArguments": []
        }
      ],
      "JLib.Data.MapDataProvider<>": [
        {
          "Lifetime": "Scoped",
          "ImplementationSource": "Type",
          "ServiceType": "JLib.Data.MapDataProvider<>",
          "ServiceArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity",
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestGraphQlDataObject"
          ],
          "ImplementationType": "JLib.Data.MapDataProvider<>",
          "ImplementationTypeArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity",
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestGraphQlDataObject"
          ]
        }
      ],
      "JLib.Data.ISourceDataProviderR<>": [
        {
          "Lifetime": "Scoped",
          "ImplementationSource": "Factory",
          "ServiceType": "JLib.Data.ISourceDataProviderR<>",
          "ServiceArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestGraphQlDataObject"
          ],
          "ImplementationType": "JLib.Data.MapDataProvider<>",
          "ImplementationTypeArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity",
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestGraphQlDataObject"
          ]
        }
      ]
    },
    "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity": {
      "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestDataProvider<>": [
        {
          "Lifetime": "Scoped",
          "ImplementationSource": "Type",
          "ServiceType": "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestDataProvider<>",
          "ServiceArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity"
          ],
          "ImplementationType": "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestDataProvider<>",
          "ImplementationTypeArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity"
          ]
        }
      ],
      "JLib.Data.IDataProviderR<>": [
        {
          "Lifetime": "Scoped",
          "ImplementationSource": "Factory",
          "ServiceType": "JLib.Data.IDataProviderR<>",
          "ServiceArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity"
          ],
          "ImplementationType": "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestDataProvider<>",
          "ImplementationTypeArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity"
          ]
        }
      ],
      "JLib.Data.IDataProviderRw<>": [
        {
          "Lifetime": "Scoped",
          "ImplementationSource": "Factory",
          "ServiceType": "JLib.Data.IDataProviderRw<>",
          "ServiceArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity"
          ],
          "ImplementationType": "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestDataProvider<>",
          "ImplementationTypeArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity"
          ]
        }
      ],
      "JLib.Data.ISourceDataProviderR<>": [
        {
          "Lifetime": "Scoped",
          "ImplementationSource": "Factory",
          "ServiceType": "JLib.Data.ISourceDataProviderR<>",
          "ServiceArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity"
          ],
          "ImplementationType": "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestDataProvider<>",
          "ImplementationTypeArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity"
          ]
        }
      ],
      "JLib.Data.ISourceDataProviderRw<>": [
        {
          "Lifetime": "Scoped",
          "ImplementationSource": "Factory",
          "ServiceType": "JLib.Data.ISourceDataProviderRw<>",
          "ServiceArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity"
          ],
          "ImplementationType": "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestDataProvider<>",
          "ImplementationTypeArguments": [
            "JLib.Tests.Reflection.ServiceCollection.AddMapDataProvider.AddMapDataProviderTests.TestXrmEntity"
          ]
        }
      ]
    }
  }
}
